File: src/types/screenshot.ts
/** 显示器信息 */
export interface MonitorInfo {
  id: number
  name: string
  x: number
  y: number
  width: number
  height: number
  isPrimary: boolean
  scaleFactor: number
}

/** 窗口信息（用于智能吸附） */
export interface WindowInfo {
  title: string
  appName: string
  x: number
  y: number
  width: number
  height: number
  isMinimized: boolean
}

/** 截图结果 */
export interface CaptureResult {
  filePath: string
  width: number
  height: number
  captureId: string
  capturedAt: string
}

/** 截图历史记录项 */
export interface ScreenshotHistoryItem {
  id: string
  filePath: string
  width: number
  height: number
  fileSize: number
  capturedAt: string
}

// ── 标注相关类型 ──────────────────────────────────────────────

/** 标注工具类型 */
export type AnnotationTool =
  | 'arrow'
  | 'rectangle'
  | 'ellipse'
  | 'text'
  | 'mosaic'
  | 'blur'
  | 'freehand'
  | 'highlight'
  | 'counter'

/** 标注样式 */
export interface AnnotationStyle {
  color: string
  strokeWidth: number
  fontSize?: number
  opacity?: number
  filled?: boolean
}

/** 点坐标 */
export interface Point {
  x: number
  y: number
}

/** 标注数据 */
export interface Annotation {
  id: string
  tool: AnnotationTool
  style: AnnotationStyle
  /** 关键点集合（箭头/画笔/高亮用） */
  points: Point[]
  /** 文字内容（文字工具用） */
  text?: string
  /** 矩形区域（矩形/椭圆/马赛克/模糊用） */
  rect?: { x: number; y: number; w: number; h: number }
  /** 序号值（序号标注用） */
  counterValue?: number
}

/** 截图模式 */
export type CaptureMode = 'fullscreen' | 'region' | 'window'

/** 编辑器模式 */
export type EditorMode = 'dashboard' | 'editor'

File: src-tauri/src/models/screenshot.rs
use serde::{Deserialize, Serialize};

/// 显示器信息
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MonitorInfo {
    pub id: u32,
    pub name: String,
    pub x: i32,
    pub y: i32,
    pub width: u32,
    pub height: u32,
    pub is_primary: bool,
    pub scale_factor: f32,
}

/// 窗口信息（用于智能吸附）
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WindowInfo {
    pub title: String,
    pub app_name: String,
    pub x: i32,
    pub y: i32,
    pub width: u32,
    pub height: u32,
    pub is_minimized: bool,
}

/// 截图结果
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CaptureResult {
    /// 临时文件路径
    pub file_path: String,
    /// 图片宽度（物理像素）
    pub width: u32,
    /// 图片高度（物理像素）
    pub height: u32,
    /// 唯一标识
    pub capture_id: String,
    /// 截图时间 ISO 8601
    pub captured_at: String,
}

/// 截图历史记录项
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ScreenshotHistoryItem {
    pub id: String,
    pub file_path: String,
    pub width: u32,
    pub height: u32,
    pub file_size: u64,
    pub captured_at: String,
}
